plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java

```diff
@@ -104,9 +109,36 @@ public ConnectorSplitSource getSplits(
         TupleDomain<ColumnHandle> tableConstraint = bigQueryTableHandle.constraint();
         Optional<String> filter = BigQueryFilterQueryBuilder.buildFilter(tableConstraint);
 
-        if (!bigQueryTableHandle.isNamedRelation()) {
+        if (bigQueryTableHandle.isQueryRelation()) {
+            BigQueryQueryRelationHandle bigQueryQueryRelationHandle = bigQueryTableHandle.getRequiredQueryRelation();
             List<BigQueryColumnHandle> columns = bigQueryTableHandle.projectedColumns().orElse(ImmutableList.of());
-            return new FixedSplitSource(BigQuerySplit.forViewStream(columns, filter));
+            TableId destinationTable = buildDestinationTable(bigQueryQueryRelationHandle.getDestinationTableName().toTableId());
+
+            String query = filter
+                    .map(whereClause -> "SELECT * FROM (" + bigQueryQueryRelationHandle.getQuery() + ") WHERE " + whereClause)
+                    .orElseGet(bigQueryQueryRelationHandle::getQuery);
+
+            try {
+                TableInfo tableInfo = new ViewMaterializationCache.DestinationTableBuilder(bigQueryClientFactory.create(session), viewExpiration, query, destinationTable).get();
+                List<String> projectedColumnsNames = getProjectedColumnNames(columns);
+
+                log.debug("Using Storage API for running query: " + query);
+                // filter is already used while constructing the select query
+                ReadSession readSession = createReadSession(session, tableInfo.getTableId(), ImmutableList.copyOf(projectedColumnsNames), Optional.empty());
+                return new FixedSplitSource(readSession.getStreamsList().stream()
+                        .map(stream -> BigQuerySplit.forStream(stream.getName(), getSchemaAsString(readSession), columns, OptionalInt.of(stream.getSerializedSize())))
+                        .collect(toImmutableList()));
+            }
+            catch (BigQueryException ex) {
+                if (ex.getMessage().startsWith("Duplicate column names in the result are not supported when a destination table is present.")) {
+                    // Use REST API for this case
+                    log.debug("Using Rest API for running query: " + query);
+                    return new FixedSplitSource(BigQuerySplit.forViewStream(columns, filter));
+                }
+                else {
+                    throw ex;
```

Comment:
Remove redundant 'else'. 

You could change IntelliJ setting with  https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#additional-ide-configuration